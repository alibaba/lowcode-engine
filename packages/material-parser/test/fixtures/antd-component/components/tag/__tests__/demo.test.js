import demoTest from '../../../tests/shared/demoTest';

demoTest('tag');
