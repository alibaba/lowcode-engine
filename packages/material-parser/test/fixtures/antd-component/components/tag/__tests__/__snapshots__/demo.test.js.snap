// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/tag/demo/animation.md correctly 1`] = `
<div>
  <div
    style="margin-bottom:16px"
  >
    <div>
      <span
        style="display:inline-block"
      >
        <span
          class="ant-tag"
        >
          Tag 1
          <span
            aria-label="close"
            class="anticon anticon-close"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        style="display:inline-block"
      >
        <span
          class="ant-tag"
        >
          Tag 2
          <span
            aria-label="close"
            class="anticon anticon-close"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        style="display:inline-block"
      >
        <span
          class="ant-tag"
        >
          Tag 3
          <span
            aria-label="close"
            class="anticon anticon-close"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </span>
      </span>
    </div>
  </div>
  <span
    class="ant-tag site-tag-plus"
  >
    <span
      aria-label="plus"
      class="anticon anticon-plus"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <defs />
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </span>
     New Tag
  </span>
</div>
`;

exports[`renders ./components/tag/demo/basic.md correctly 1`] = `
<div>
  <span
    class="ant-tag"
  >
    Tag 1
  </span>
  <span
    class="ant-tag"
  >
    <a
      href="https://github.com/ant-design/ant-design/issues/1862"
    >
      Link
    </a>
  </span>
  <span
    class="ant-tag"
  >
    Tag 2
    <span
      aria-label="close"
      class="anticon anticon-close"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="close"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </span>
  </span>
  <span
    class="ant-tag"
  >
    Prevent Default
    <span
      aria-label="close"
      class="anticon anticon-close"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="close"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`renders ./components/tag/demo/checkable.md correctly 1`] = `
<div>
  <span
    class="ant-tag ant-tag-checkable ant-tag-checkable-checked"
  >
    Tag1
  </span>
  <span
    class="ant-tag ant-tag-checkable ant-tag-checkable-checked"
  >
    Tag2
  </span>
  <span
    class="ant-tag ant-tag-checkable ant-tag-checkable-checked"
  >
    Tag3
  </span>
</div>
`;

exports[`renders ./components/tag/demo/colorful.md correctly 1`] = `
<div>
  <h4
    style="margin-bottom:16px"
  >
    Presets:
  </h4>
  <div>
    <span
      class="ant-tag ant-tag-magenta"
    >
      magenta
    </span>
    <span
      class="ant-tag ant-tag-red"
    >
      red
    </span>
    <span
      class="ant-tag ant-tag-volcano"
    >
      volcano
    </span>
    <span
      class="ant-tag ant-tag-orange"
    >
      orange
    </span>
    <span
      class="ant-tag ant-tag-gold"
    >
      gold
    </span>
    <span
      class="ant-tag ant-tag-lime"
    >
      lime
    </span>
    <span
      class="ant-tag ant-tag-green"
    >
      green
    </span>
    <span
      class="ant-tag ant-tag-cyan"
    >
      cyan
    </span>
    <span
      class="ant-tag ant-tag-blue"
    >
      blue
    </span>
    <span
      class="ant-tag ant-tag-geekblue"
    >
      geekblue
    </span>
    <span
      class="ant-tag ant-tag-purple"
    >
      purple
    </span>
  </div>
  <h4
    style="margin:16px 0"
  >
    Custom:
  </h4>
  <div>
    <span
      class="ant-tag ant-tag-has-color"
      style="background-color:#f50"
    >
      #f50
    </span>
    <span
      class="ant-tag ant-tag-has-color"
      style="background-color:#2db7f5"
    >
      #2db7f5
    </span>
    <span
      class="ant-tag ant-tag-has-color"
      style="background-color:#87d068"
    >
      #87d068
    </span>
    <span
      class="ant-tag ant-tag-has-color"
      style="background-color:#108ee9"
    >
      #108ee9
    </span>
  </div>
</div>
`;

exports[`renders ./components/tag/demo/control.md correctly 1`] = `
<div>
  <span
    class="ant-tag edit-tag"
  >
    <span>
      Unremovable
    </span>
  </span>
  <span
    class="ant-tag edit-tag"
  >
    <span>
      Tag 2
    </span>
    <span
      aria-label="close"
      class="anticon anticon-close"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="close"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </span>
  </span>
  <span
    class="ant-tag edit-tag"
  >
    <span>
      Tag 3
    </span>
    <span
      aria-label="close"
      class="anticon anticon-close"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="close"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </span>
  </span>
  <span
    class="ant-tag site-tag-plus"
  >
    <span
      aria-label="plus"
      class="anticon anticon-plus"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <defs />
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </span>
     New Tag
  </span>
</div>
`;

exports[`renders ./components/tag/demo/controlled.md correctly 1`] = `
<div>
  <span
    class="ant-tag"
  >
    Movies
    <span
      aria-label="close"
      class="anticon anticon-close"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="close"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </span>
  </span>
  <br />
  <button
    class="ant-btn ant-btn-sm"
    type="button"
  >
    <span>
      Toggle
    </span>
  </button>
</div>
`;

exports[`renders ./components/tag/demo/hot-tags.md correctly 1`] = `
<div>
  <span
    style="margin-right:8px"
  >
    Categories:
  </span>
  <span
    class="ant-tag ant-tag-checkable"
  >
    Movies
  </span>
  <span
    class="ant-tag ant-tag-checkable"
  >
    Books
  </span>
  <span
    class="ant-tag ant-tag-checkable"
  >
    Music
  </span>
  <span
    class="ant-tag ant-tag-checkable"
  >
    Sports
  </span>
</div>
`;

exports[`renders ./components/tag/demo/icon.md correctly 1`] = `
<div>
  <span
    class="ant-tag ant-tag-has-color"
    style="background-color:#55acee"
  >
    <span
      aria-label="twitter"
      class="anticon anticon-twitter"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="twitter"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0075-94 336.64 336.64 0 01-108.2 41.2A170.1 170.1 0 00672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 00-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 01-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 01-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
        />
      </svg>
    </span>
    <span>
      Twitter
    </span>
  </span>
  <span
    class="ant-tag ant-tag-has-color"
    style="background-color:#cd201f"
  >
    <span
      aria-label="youtube"
      class="anticon anticon-youtube"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="youtube"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M960 509.2c0-2.2 0-4.7-.1-7.6-.1-8.1-.3-17.2-.5-26.9-.8-27.9-2.2-55.7-4.4-81.9-3-36.1-7.4-66.2-13.4-88.8a139.52 139.52 0 00-98.3-98.5c-28.3-7.6-83.7-12.3-161.7-15.2-37.1-1.4-76.8-2.3-116.5-2.8-13.9-.2-26.8-.3-38.4-.4h-29.4c-11.6.1-24.5.2-38.4.4-39.7.5-79.4 1.4-116.5 2.8-78 3-133.5 7.7-161.7 15.2A139.35 139.35 0 0082.4 304C76.3 326.6 72 356.7 69 392.8c-2.2 26.2-3.6 54-4.4 81.9-.3 9.7-.4 18.8-.5 26.9 0 2.9-.1 5.4-.1 7.6v5.6c0 2.2 0 4.7.1 7.6.1 8.1.3 17.2.5 26.9.8 27.9 2.2 55.7 4.4 81.9 3 36.1 7.4 66.2 13.4 88.8 12.8 47.9 50.4 85.7 98.3 98.5 28.2 7.6 83.7 12.3 161.7 15.2 37.1 1.4 76.8 2.3 116.5 2.8 13.9.2 26.8.3 38.4.4h29.4c11.6-.1 24.5-.2 38.4-.4 39.7-.5 79.4-1.4 116.5-2.8 78-3 133.5-7.7 161.7-15.2 47.9-12.8 85.5-50.5 98.3-98.5 6.1-22.6 10.4-52.7 13.4-88.8 2.2-26.2 3.6-54 4.4-81.9.3-9.7.4-18.8.5-26.9 0-2.9.1-5.4.1-7.6v-5.6zm-72 5.2c0 2.1 0 4.4-.1 7.1-.1 7.8-.3 16.4-.5 25.7-.7 26.6-2.1 53.2-4.2 77.9-2.7 32.2-6.5 58.6-11.2 76.3-6.2 23.1-24.4 41.4-47.4 47.5-21 5.6-73.9 10.1-145.8 12.8-36.4 1.4-75.6 2.3-114.7 2.8-13.7.2-26.4.3-37.8.3h-28.6l-37.8-.3c-39.1-.5-78.2-1.4-114.7-2.8-71.9-2.8-124.9-7.2-145.8-12.8-23-6.2-41.2-24.4-47.4-47.5-4.7-17.7-8.5-44.1-11.2-76.3-2.1-24.7-3.4-51.3-4.2-77.9-.3-9.3-.4-18-.5-25.7 0-2.7-.1-5.1-.1-7.1v-4.8c0-2.1 0-4.4.1-7.1.1-7.8.3-16.4.5-25.7.7-26.6 2.1-53.2 4.2-77.9 2.7-32.2 6.5-58.6 11.2-76.3 6.2-23.1 24.4-41.4 47.4-47.5 21-5.6 73.9-10.1 145.8-12.8 36.4-1.4 75.6-2.3 114.7-2.8 13.7-.2 26.4-.3 37.8-.3h28.6l37.8.3c39.1.5 78.2 1.4 114.7 2.8 71.9 2.8 124.9 7.2 145.8 12.8 23 6.2 41.2 24.4 47.4 47.5 4.7 17.7 8.5 44.1 11.2 76.3 2.1 24.7 3.4 51.3 4.2 77.9.3 9.3.4 18 .5 25.7 0 2.7.1 5.1.1 7.1v4.8zM423 646l232-135-232-133z"
        />
      </svg>
    </span>
    <span>
      Youtube
    </span>
  </span>
  <span
    class="ant-tag ant-tag-has-color"
    style="background-color:#3b5999"
  >
    <span
      aria-label="facebook"
      class="anticon anticon-facebook"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="facebook"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-32 736H663.9V602.2h104l15.6-120.7H663.9v-77.1c0-35 9.7-58.8 59.8-58.8h63.9v-108c-11.1-1.5-49-4.8-93.2-4.8-92.2 0-155.3 56.3-155.3 159.6v89H434.9v120.7h104.3V848H176V176h672v672z"
        />
      </svg>
    </span>
    <span>
      Facebook
    </span>
  </span>
  <span
    class="ant-tag ant-tag-has-color"
    style="background-color:#55acee"
  >
    <span
      aria-label="linkedin"
      class="anticon anticon-linkedin"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="linkedin"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M847.7 112H176.3c-35.5 0-64.3 28.8-64.3 64.3v671.4c0 35.5 28.8 64.3 64.3 64.3h671.4c35.5 0 64.3-28.8 64.3-64.3V176.3c0-35.5-28.8-64.3-64.3-64.3zm0 736c-447.8-.1-671.7-.2-671.7-.3.1-447.8.2-671.7.3-671.7 447.8.1 671.7.2 671.7.3-.1 447.8-.2 671.7-.3 671.7zM230.6 411.9h118.7v381.8H230.6zm59.4-52.2c37.9 0 68.8-30.8 68.8-68.8a68.8 68.8 0 10-137.6 0c-.1 38 30.7 68.8 68.8 68.8zm252.3 245.1c0-49.8 9.5-98 71.2-98 60.8 0 61.7 56.9 61.7 101.2v185.7h118.6V584.3c0-102.8-22.2-181.9-142.3-181.9-57.7 0-96.4 31.7-112.3 61.7h-1.6v-52.2H423.7v381.8h118.6V604.8z"
        />
      </svg>
    </span>
    <span>
      LinkedIn
    </span>
  </span>
</div>
`;

exports[`renders ./components/tag/demo/status.md correctly 1`] = `
<div>
  <span
    class="ant-tag ant-tag-success"
  >
    success
  </span>
  <span
    class="ant-tag ant-tag-processing"
  >
    processing
  </span>
  <span
    class="ant-tag ant-tag-error"
  >
    error
  </span>
  <span
    class="ant-tag ant-tag-default"
  >
    default
  </span>
  <span
    class="ant-tag ant-tag-warning"
  >
    warning
  </span>
</div>
`;
